### ISSUE TITLE
how to add  --privilleged for docker when run cromwell in aws

### ISSUE THREAD

**Issue by openbioinfo** [broadinstitute/cromwell#5863]
how to add  --privilleged for docker when run cromwell in aws

where to add this option ?

**Comment by markjschreiber** [comment]
There’s currently no way to do this. It would need to be added to the part
of the code that constructs the job definition.

What is the scenario requiring running as privileged?

**Comment by openbioinfo** [comment]
i plan to run singularity images in docker container.

**Comment by guma44** [comment]
Did you try to implement your own docker-submit in the config file?

**Comment by openbioinfo** [comment]
no.  in aws mode, i dont find docker-submit.  

i prefer `--privileged=true` as default for cromwell with aws as backend.

**Comment by guma44** [comment]
"submit-docker" (sorry for reversal) is one of the configuration option in Cromwell config file. See eg. here for additional volumes mount (last section): https://davetang.org/muse/2019/12/24/execute-gatk-workflows-locally

**Comment by openbioinfo** [comment]
thanks for you reply.

i mean in `aws backend ` mode, instead of `local mode`.  there is no option to set `submit-docker`, i attached the backend part of my aws.conf as follows.

```bash
backend {
     default = "AWSBATCH"
     providers {
         AWSBATCH {
             actor-factory = "cromwell.backend.impl.aws.AwsBatchBackendLifecycleActorFactory"
             config {
                 // Base bucket for workflow executions
                 root = "s3://yuce/cromwell-execution"
                 // A reference to an auth defined in the `aws` stanza at the top. This auth is used to create
                 // Jobs and manipulate auth JSONs.
                 auth = "default"

                 numSubmitAttempts = 3
                 numCreateDefinitionAttempts = 3

                 concurrent-job-limit = 16

                 default-runtime-attributes {
                    queueArn: "arn:aws-cn:batch:cn-northwest-1:723230375162:job-queue/first-run-job-queue",
                 }

                 filesystems {
                     s3 {
                         // A reference to a potentially different auth for manipulating files via engine functions.
                         auth = "default"
                     }
                 }
             }
         }
     }
}

```

**Comment by guma44** [comment]
By "no option to set" do you mean that is not allowed to do this? Cannot you just add this section? Local is just an example. I have UGE backend where I override "submit-docker" to use it with singularity.

**Comment by openbioinfo** [comment]
yes. i hava tried. in aws mode, cromwell does not support `submit-docker`. 

https://github.com/broadinstitute/cromwell/issues/5863#issuecomment-699685666

**Comment by guma44** [comment]
Ah, OK. This looks like something to add.

**Comment by scanon** [comment]
We have a similar need.  This overlaps a little with #4579.  It would be useful if the submit-docker was parameterized similar to how it is for some of the other backends.

**Comment by markjschreiber** [comment]
Thanks for the feedback. Can you elaborate more on the need to be able to
run a container as privileged?

It could (in theory) be parameterized if required but it seems hazardous to
have this be the default.

**Comment by leipzig** [comment]
One scenario that comes to mind is running anything that requires a docker or singularity server, such as Nextflow. So if you need to run Nextflow as a step in a larger WDL workflow it is going to want to pull images
